modules/apps/29-fee/ibc_module_test.go

```diff
@@ -468,3 +469,331 @@ func (suite *FeeTestSuite) TestOnChanCloseConfirm() {
+					ForwardRelayerAddress: suite.chainB.SenderAccount.GetAddress().String(),
+				}
+				suite.Require().Equal(result, ack)
+			}
+			if !tc.sourceRelayer && tc.feeEnabled {
+				suite.Require().NoError(err, "unexpected error for case: %s", tc.name)
+				ack := types.IncentivizedAcknowledgement{
+					Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+					ForwardRelayerAddress: "",
+				}
+				suite.Require().Equal(result, ack)
+			}
+			if tc.sourceRelayer && !tc.feeEnabled {
+				suite.Require().NoError(err, "unexpected error for case: %s", tc.name)
+				ack := channeltypes.NewResultAcknowledgement([]byte{1})
+				suite.Require().Equal(result, ack)
+			}
+		})
+	}
+}
+
+func (suite *FeeTestSuite) TestOnAcknowledgementPacket() {
+	testCases := []struct {
+		name     string
+		malleate func()
+		ackBytes []byte
+		expPass  bool
+	}{
+		{
+			"success",
+			func() {
+				packetId := channeltypes.NewPacketId(suite.path.EndpointA.ChannelID, suite.path.EndpointA.ChannelConfig.PortID, suite.chainA.SenderAccount.GetSequence())
+				identifiedFee := types.NewIdentifiedPacketFee(
+					packetId,
+					types.Fee{
+						ReceiveFee: validCoins,
+						AckFee:     validCoins2,
+						TimeoutFee: validCoins3,
+					},
+					suite.chainA.SenderAccount.GetAddress().String(),
+					[]string{},
+				)
+				err := suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), identifiedFee)
+				suite.Require().NoError(err)
+			},
+			types.IncentivizedAcknowledgement{
+				Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+				ForwardRelayerAddress: suite.chainB.SenderAccount.GetAddress().String(),
+			}.Acknowledgement(),
+			true,
+		},
+		{
+			"fee not enabled, success",
+			func() {
+				suite.chainA.GetSimApp().IBCFeeKeeper.DeleteFeeEnabled(suite.chainA.GetContext(), suite.path.EndpointA.ChannelConfig.PortID, suite.path.EndpointA.ChannelID)
+			},
+			channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+			true,
+		},
+		{
+			"fee not enabled, fail",
+			func() {
+				suite.chainA.GetSimApp().IBCFeeKeeper.DeleteFeeEnabled(suite.chainA.GetContext(), suite.path.EndpointA.ChannelConfig.PortID, suite.path.EndpointA.ChannelID)
+			},
+			types.IncentivizedAcknowledgement{
+				Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+				ForwardRelayerAddress: suite.chainB.SenderAccount.GetAddress().String(),
+			}.Acknowledgement(),
+			false,
+		},
+		{
+			"identified packet fee doesn't exist",
+			func() {
+				suite.Require().Error(types.ErrFeeNotFound)
+			},
+			types.IncentivizedAcknowledgement{
+				Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+				ForwardRelayerAddress: suite.chainB.SenderAccount.GetAddress().String(),
+			}.Acknowledgement(),
+			false,
+		},
+		{
+			"ack wrong format",
```

Comment:
this appears not to be testing the lines as expected. I see missing code coverage for unmarshling error of unmarshalling into incentivized ack. Maybe try making the ack be some random bytes like []byte("not a valid acknowledgement")